pinot-core/src/main/java/com/linkedin/pinot/core/startree/DimensionBuffer.java

```diff
@@ -15,47 +15,42 @@
  */
 package com.linkedin.pinot.core.startree;
 
-import com.linkedin.pinot.core.segment.creator.impl.V1Constants;
+import com.linkedin.pinot.core.segment.memory.PinotDataBuffer;
 import java.nio.ByteBuffer;
-import java.nio.ByteOrder;
-import java.nio.IntBuffer;
 
 
 public class DimensionBuffer {
-  private static final ByteOrder NATIVE_ORDER = ByteOrder.nativeOrder();
+  private final ByteBuffer _buffer;
 
-  private final int _numDimensions;
-  private final IntBuffer _intBuffer;
+  public static DimensionBuffer fromBytes(byte[] bytes) {
+    return new DimensionBuffer(bytes);
+  }
 
   public DimensionBuffer(int numDimensions) {
-    _numDimensions = numDimensions;
-    _intBuffer = ByteBuffer.wrap(new byte[numDimensions * V1Constants.Numbers.INTEGER_SIZE]).asIntBuffer();
+    _buffer = ByteBuffer.wrap(new byte[numDimensions * Integer.BYTES]).order(PinotDataBuffer.NATIVE_ORDER);
   }
 
-  public DimensionBuffer(byte[] bytes) {
-    _numDimensions = bytes.length / V1Constants.Numbers.INTEGER_SIZE;
-    // NOTE: the byte array is returned from Unsafe which uses native order
-    _intBuffer = ByteBuffer.wrap(bytes).order(NATIVE_ORDER).asIntBuffer();
+  private DimensionBuffer(byte[] bytes) {
+    _buffer = ByteBuffer.wrap(bytes).order(PinotDataBuffer.NATIVE_ORDER);
   }
 
-  public static DimensionBuffer fromBytes(byte[] bytes) {
-    return new DimensionBuffer(bytes);
+  public int getDictId(int index) {
+    return _buffer.getInt(index * Integer.BYTES);
   }
 
-  public int getDimension(int index) {
-    return _intBuffer.get(index);
+  public void setDictId(int index, int dictId) {
+    _buffer.putInt(index * Integer.BYTES, dictId);
   }
 
-  public void setDimension(int index, int value) {
-    _intBuffer.put(index, value);
+  public byte[] toBytes() {
+    return _buffer.array();
   }
 
-  @SuppressWarnings({"EqualsWhichDoesntCheckParameterClass", "CheckStyle"})
-  @Override
-  public boolean equals(Object obj) {
-    DimensionBuffer that = (DimensionBuffer) obj;
-    for (int i = 0; i < _numDimensions; i++) {
-      if (_intBuffer.get(i) != that._intBuffer.get(i)) {
+  public boolean hasSameBytes(byte[] dimensionBytes) {
```

Comment:
If the assumption is that both the buffers should be of same length, please document it.

Reply:
Added.